ipv8/test/mocking/rest/rest_api_peer.py

```diff
@@ -0,0 +1,324 @@
+        """
+        Add a new verified peer
+
+        :param peer: the new peer
+        :return: None
+        """
+        self._ipv8.network.add_verified_peer(peer)
+
+    def add_peer_to_all_services(self, peer):
+        """
+        Add a pier to the identity service
+
+        :param peer: the peer to be added to the identity service
+        :return: None
+        """
+        self._ipv8.network.discover_services(peer, [overlay.master_peer.mid for overlay in self._ipv8.overlays])
+
+    def add_peer_address(self, peer, interface, port):
+        """
+        Add the address of a peer so it becomes accessible
+
+        :param peer: the peer whose address will be added
+        :param interface: The interface (IP or alias) of the peer
+        :param port: The port on which the peer accepts requests
+        :return: None
+        """
+        self._ipv8.network.discover_address(peer, (interface, port))
+
+    def close(self):
+        """
+        Stop the peer
+
+        :return: None
+        """
+        self._logger.info("Shutting down the peer")
+
+        self._rest_manager.stop()
+        self._rest_manager.shutdown_task_manager()
+
+        self._ipv8.unload()
+
+    def get_keys(self):
+        """
+        Get the peer's keys
+
+        :return: the peer's keys
+        """
+        self._logger.info("Fetching my IPv8 object's peer.")
+        return self._ipv8.keys
+
+    def get_mids(self, replace_characters=True):
+        """
+        Return a list of the b64 encoded mids of this peer
+
+        :param replace_characters: a boolean variable, which indicates whether certain characters which cannot
+                                  be forwarded within an HTTP request should be replaced
+        :return: a list of the peer's mids (encoded in b64)
+        """
+        if replace_characters:
+            return [b64encode(x.mid) for x in self._ipv8.keys.values()]
+
+        return [b64encode(x.mid) for x in self._ipv8.keys.values()]
+
+    def get_mid_by_key(self, key='my peer'):
+        """
+        Return a mid given a key which identifies one of this peer's identity
+
+        :param key: the key supplied. By default 'my peer'
+        :return: the chosen mid
+        """
+        return string_to_url(b64encode(self._ipv8.keys[key].mid)) if key in self._ipv8.keys else ''
+
+    def get_overlay_by_class(self, name):
+        """
+        Get one of the peer's overlays as identified by its name
+
+        :parameter name: the name of the overlay
+        :return: the peer's overlays
+        """
+        self._logger.info("Fetching my IPv8 object's overlay: %s", name)
+        for overlay in self._ipv8.overlays:
+            if isinstance(overlay, name):
+                return overlay
+
+        return None
+
+    @property
+    def port(self):
+        return self._port
+
+    @property
+    def interface(self):
+        return self._interface
+
+    class RestAPITestWrapper(TaskManager):
```

Comment:
Wanted to reduce its scope, since it seems to be very particular to the tests concerning the REST API, hence it only gets used by the `TestPeer` class.